src/kubernetes/client/config/in_cluster.rs

```diff
@@ -0,0 +1,75 @@
+//! Everything related to building in-cluster configuration.
+
+use super::Config;
+use crate::tls::TlsOptions;
+use http02::Uri;
+use snafu::{ResultExt, Snafu};
+
+impl Config {
+    /// Prepares a config suitable for use when running in k8s cluster.
+    pub fn in_cluster() -> Result<Self, Error> {
+        let host = std::env::var("KUBERNETES_SERVICE_HOST").context(NotInCluster {
+            missing: "KUBERNETES_SERVICE_HOST",
+        })?;
+        let port = std::env::var("KUBERNETES_SERVICE_PORT").context(NotInCluster {
+            missing: "KUBERNETES_SERVICE_PORT",
+        })?;
+
+        let base = Uri::builder()
+            .scheme("https")
+            .authority(join_host_port(host.as_str(), port.as_str()).as_str())
+            .path_and_query("/")
+            .build()
+            .context(InvalidUrl)?;
+
+        let token_file = "/var/run/secrets/kubernetes.io/serviceaccount/token";
```

Comment:
How does this work on Windows?

Reply:
No, see https://github.com/timberio/vector/pull/2653#discussion_r432031705